pandas/core/categorical.py

```diff
@@ -1995,14 +1995,18 @@ def is_dtype_equal(self, other):
 
         Parameters
         ----------
-        other : Categorical
+        other : Categorical, Series
 
         Returns
         -------
         are_equal : boolean
         """
-
         try:
+            from pandas.core.series import Series
+
+            if isinstance(other, Series):
```

Comment:
Thank you for the advice. However, this seems not to work either.

It will at least make the following test case to fail,
```python
assert c3.is_dtype_equal(c3)
```
One of the reason is that the new instance created by `Categorical(c3)` will not preserve the attribute `ordered` of `c3`. Not quite sure whether this is by design.